### PR TITLE
samples: add event time as a custom attribute

### LINKED ISSUE #856: Need a more obvious way to set event time in a message
`EventTime` is useful in deciding when to advance high watermarks for streaming engines (Beam/Spark/Flink). Right now to set event time in a message published to Pub/Sub Lite, one needs to use a custom attribute of key "x-goog-pubsublite-event-time-timestamp-proto" and pass in a string using `MessageTransforms.encodeAttributeEventTime()`:

```java
import com.google.cloud.pubsublite.cloudpubsub.MessageTransforms;

// ..

    String eventTime = MessageTransforms
        .encodeAttributeEventTime(Timestamp.newBuilder().setSeconds(100).build());

    ByteString data = ByteString.copyFromUtf8(message);

    PubsubMessage pubsubMessage =
        PubsubMessage.newBuilder()
            .setData(data)
            .putAttributes("x-goog-pubsublite-event-time-timestamp-proto", eventTime)
            .build();

    ApiFuture<String> future = publisher.publish(pubsubMessage);
```

Is there a better way to surface `EventTime` from the protos?

https://github.com/googleapis/java-pubsublite/blob/343afbffc2be92cb6be2670ef3ec34c599cd95dc/proto-google-cloud-pubsublite-v1/src/main/java/com/google/cloud/pubsublite/proto/PubSubMessageOrBuilder.java#L114-L144

### PR DESCRIPTION
**Pull request by anguillanneuf**
Fixes #856 

Show the proper way to publish a message to Pub/Sub Lite with an event time as an attribute.

### CODE REVIEW

File: samples/snippets/src/main/java/pubsublite/PublishWithCustomAttributesExample.java [hunk 1]
```diff
@@ -64,15 +67,23 @@ public static void publishWithCustomAttributesExample(
...
-    String message = "message-with-custom-attributes";
+    // Prepare the message data as a byte string.
+    String messageData = "message-with-custom-attributes";
+    ByteString data = ByteString.copyFromUtf8(messageData);
+
+    // Prepare a protobuf-encoded event timestamp for the message.
+    Instant now = Instant.now();
+    String eventTime =
+        MessageTransforms.encodeAttributeEventTime(
+            Timestamp.newBuilder().setSeconds(now.getEpochSecond()).build());
```
Comment (dpcollins-google) [inline samples/snippets/src/main/java/pubsublite/PublishWithCustomAttributesExample.java:78]: use https://developers.google.com/protocol-buffers/docs/reference/java/com/google/protobuf/util/Timestamps

`Timestamps.fromMillis(now.toEpochMilli())`


toEpochMilli()
--





toEpochMilli()
--

File: samples/snippets/src/main/java/pubsublite/SubscriberExample.java [hunk 2]
```diff
@@ -71,6 +71,12 @@ public static void subscriberExample(
         (PubsubMessage message, AckReplyConsumer consumer) -> {
           System.out.println("Id : " + MessageMetadata.decode(message.getMessageId()));
           System.out.println("Data : " + message.getData().toStringUtf8());
+          System.out.println("Ordering key : " + message.getOrderingKey());
+          System.out.println("Attributes : ");
+          message
```
Comment (dpcollins-google) [inline samples/snippets/src/main/java/pubsublite/SubscriberExample.java:76]: I'd also demonstrate extracting the event time

MessageTransforms. decodeAttributeEventTime(message.getAttributesMap().get(MessageTransforms.PUBSUB_LITE_EVENT_TIME_TIMESTAMP_PROTO))

Comment (anguillanneuf) [reply samples/snippets/src/main/java/pubsublite/SubscriberExample.java:76]: This is a generic SubscriberExample for messages with and without attributes.

File: samples/snippets/src/main/java/pubsublite/PublishWithCustomAttributesExample.java (hunk 1 above)
Comment (anguillanneuf) [reply samples/snippets/src/main/java/pubsublite/PublishWithCustomAttributesExample.java]: That's better. Udpated.

File: samples/snippets/src/main/java/pubsublite/SubscriberExample.java (hunk 2 above)
Comment (dpcollins-google) [reply samples/snippets/src/main/java/pubsublite/SubscriberExample.java:78]: Ah okay. Thats fine then. Could you add:

```
if (message.getAttributesMap().containsKey(MessageTransforms.PUBSUB_LITE_EVENT_TIME_TIMESTAMP_PROTO)) {
  System.out.println("Event timestamp : ", + MessageTransforms. decodeAttributeEventTime(message.getAttributesMap().get(MessageTransforms.PUBSUB_LITE_EVENT_TIME_TIMESTAMP_PROTO)));
}
```

**Review by dpcollins-google — APPROVED** [review]

File: samples/snippets/src/main/java/pubsublite/SubscriberExample.java (hunk 2 above)
Comment (anguillanneuf) [reply samples/snippets/src/main/java/pubsublite/SubscriberExample.java:78]: Yep, a `if..else` exists with in the last commit.

**Pull request merged**